lib/engine/game/g_1828/game.rb

```diff
@@ -1513,6 +1513,18 @@ def system_market_price(corporations)
 
           new_market_price
         end
+
+        def check_other(route)
+          if route_includes_coalfields?(route.routes) && !@game.coal_marker?(current_entity)
+            raise GameError, 'Cannot run to Virginia Coalfields without a Coal Marker'
+          end
+
+          super
+        end
+
+        def route_includes_coalfields?(routes)
+          routes.flat_map(&:connection_hexes).flatten.include?(VA_COALFIELDS_HEX)
```

Comment:
it's gonna be more efficient probably if you just something like

routes.any? do | route|
  route.connection_hexes.include?(VA...)

Reply:
Ok.  I was also thinking I should only test the one route - I'll change that too.

Reply:
connection_hexes contains things like this
   0: (2) ['G23', 'G21']
so I think the flatten is needed